### PR TITLE
Switch ViewPanel base class to Grid

### LINKED ISSUE #8420: Views with borderRadius apply transform to nested views twice
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Add the following example to RNTester's TransformExample:
```js
  {
    title: 'Translate with corner radius',
    description: 'Translate with corner radius',
    render(): Node {
      return (
        <View style={styles.container}>
          <View
            style={{
              backgroundColor: 'blue',
              borderRadius: 4,
              width: 100,
              height: 100,
              alignItems: 'center',
              justifyContent: 'center',
              transform: [{translateX: 50}],
            }}>
            <View
              style={{
                backgroundColor: 'cyan',
                width: 50,
                height: 50,
                alignItems: 'center',
                justifyContent: 'center',
              }}>
              <Text style={{color: 'white'}}>X</Text>
            </View>
          </View>
        </View>
      );
    },
  },
```
2.

## Expected Results
I expect the transform behavior to match what occurs when I do not set the `borderRadius`, which does not apply the transform to children twice.

## Snack, code example, screenshot, or link to a repository:
See example code above, which can be easily injected into any playground app or RNTester.

Here is what a transform with child views looks like without a `borderRadius`:
image

Here is what the transform looks like when a `borderRadius` is applied to the same view that also has a `transform`:
image

Another interesting issue, possibly unrelated, is that setting `borderRadius` seems to also implement `overflow: hidden` behavior. I suspect the `borderRadius` implementation just creates a custom clipping geometry.

Please note, this applies to any type of transform, so if I set `scaleX` to 2, I expect the outer and inner views to maintain the same width ratio, however the inner view doubles the scale and ends up matching the width of the outer view:
image

### LINKED ISSUE #9767: Applying a border radius to a view breaks zIndex
### Problem Description

When a view has a child view that is absolute and can sit outside of the parent view, if the parent view has borderRadius set then the child view loses the ability to float outside of the parent view.

### Steps To Reproduce

Create a view with a border radius and then inside that view, create an element with an absolute position that should sit outside of the parent view. Something like the following:

```
<View style={{borderRadius: 10}}>
    <View style={{position: 'absolute', top: -20}}></View>
</View>
```

Workaround:
```
<View style={{height: 10, borderTopLeftRadius: 10, borderTopRightRadius: 10}}></View>
<View style=={{flex: 1}}>
    <View style={{position: 'absolute', top: -20}}></View>
</View>
<View style={{height: 10, borderBottomLeftRadius: 10, borderBottomRightRadius: 10}}></View>
```

### Expected Results

Child element with absolute positioning should be able to float outside of the parent regardless of if the parent has a radius or not.

### CLI version

npx react-native --version

### Environment

```markdown
OS: Windows 10 10.0.22000
CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
Memory: 21.85 GB / 31.73 GB
Binaries:
Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
npm: 8.1.3 - C:\Program Files\nodejs\npm.CMD
Watchman: Not Found
SDKs:
Android SDK: Not Found
Windows SDK:
AllowDevelopmentWithoutDevLicense: Enabled
AllowAllTrustedApps: Enabled
Versions: 10.0.18362.0, 10.0.19041.0, 10.0.22000.0
IDEs:
Android Studio: Not Found
Visual Studio: 17.0.31912.275 (Visual Studio Community 2022)
Languages:
Java: Not Found
npmPackages:
@react-native-community/cli: Not Found
react: 17.0.2 => 17.0.2
react-native: 0.66.0 => 0.66.0
react-native-windows: 0.66.5 => 0.66.5
npmGlobalPackages:
react-native: Not Found
```

### Target Platform Version

_No response_

### Target Device(s)

_No response_

### Visual Studio Version

_No response_

### Build Configuration

_No response_

### Snack, code example, screenshot, or link to a repository

_No response_

### PR DESCRIPTION
**Pull request by rozele**
### Type of Change
_Erase all that don't apply._
- Bug fix
- New feature (non-breaking change which adds functionality)

### Why
Grid supports border properties (BorderThickness, BorderBrush, CornerRadius) intrinsically.

We experimented with using Grid as the base class for ViewPanel in Messenger and saw no regression in memory usage, and a moderate improvement in time-to-rendered-content for important scenarios (as much as 3% improvement in some cases).

In addition to the performance improvement, avoiding the complicated outer / inner border logic in ViewPanel allows us to remove over 250 lines of code (more if we could make breaking changes to the ViewPanel ABI) in ViewViewManager and ViewPanel.

Lastly, using Grid as the base class for ViewPanel allows third-party view managers to create View-like custom native components that also support border properties, without needing to rewrite the complex inner/outer border management logic.

Resolves #8420
Resolves #9767

### What
This change moves the ViewPanel base class from a base XAML Panel to XAML Grid.
 
## Testing

Before:

https://user-images.githubusercontent.com/1106239/210821491-fdaf81cc-771a-45aa-a7af-2615674e452a.mp4

After:

https://user-images.githubusercontent.com/1106239/210821930-ecc4cc78-fb1e-4fae-a96e-68597a9ec941.mp4

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/11056)

### CODE REVIEW

**Comment by asklar** [comment]
@rozele @jonthysell  changing the class hierarchy is a breaking change, we should tread carefully here.
we should make sure this isn't breaking modules that may have made assumptions here like RNX, or 1st party apps.

**Comment by rozele** [comment]
> @rozele @jonthysell changing the class hierarchy is a breaking change, we should tread carefully here.
we should make sure this isn't breaking modules that may have made assumptions here like RNX, or 1st party apps.

Indeed. Fortunately it's not a significant breaking change, since Grid still derives from Panel, so ViewPanel still has Panel as an ancestor. That being said, ViewPanel is exposed in the public ABI, so having things like GetOuterBorder return null is definitely a breaking change. We could remove GetOuterBorder to make it very explicit that we're making a breaking change here...

**Comment by rozele** [comment]
I can't figure out how to update the snapshots based on the results of this CI run. It looks like the snapshots that got uploaded are just the snapshots already committed in the repo (rather than the snapshots that were computed in this test run :( ). cc @jonthysell

**Comment by rozele** [comment]
A side benefit to this, here is what the Mouse Example renders to on other platforms:
image

Notice that the overlay view renders on top of the border.

With our current border implementation for View (with the inner/outer border approach), the border renders on top of overlay children 😬:
image

When we switch to using Grid as a base class, this issue is fixed.

File: packages/e2e-test-app/test/__snapshots__/XAMLTest.test.ts.snap
Comment (rozele) [inline packages/e2e-test-app/test/__snapshots__/XAMLTest.test.ts.snap:12]: Need to debug why these sizes changed :(

File: packages/e2e-test-app/test/__snapshots__/ViewComponentTest.test.ts.snap
Comment (rozele) [inline packages/e2e-test-app/test/__snapshots__/ViewComponentTest.test.ts.snap:371]: I think the fact that we're setting a default border brush in ViewViewManager adds a bunch of these deltas. Not clear whether it's harmful for us to do this or not :|

File: packages/e2e-test-app/test/__snapshots__/ViewComponentTest.test.ts.snap
Comment (rozele) [inline packages/e2e-test-app/test/__snapshots__/ViewComponentTest.test.ts.snap:470]: This is causing a bit of weirdness. It appears that Grid snaps it's BorderThickness to the logical pixel grid, whereas Border did not do this. I'll re-run the snapshots at 100% scaling to see if these go away, but even still, we should investigate (I have noticed that there are some trivial changes to layout due to differences in actual sizes for the thickness.

File: packages/e2e-test-app/test/__snapshots__/ViewComponentTest.test.ts.snap
Comment (rozele) [inline packages/e2e-test-app/test/__snapshots__/ViewComponentTest.test.ts.snap:1076]: Grid has a `Padding` property, where as the base Panel derived ViewPanel did not, which is why we see these Padding values.

File: packages/e2e-test-app/test/__snapshots__/PressableComponentTest.test.ts.snap
Comment (rozele) [inline packages/e2e-test-app/test/__snapshots__/PressableComponentTest.test.ts.snap:11]: We should resolve why these Top and Left values are different.

File: packages/e2e-test-app/test/__snapshots__/PressableComponentTest.test.ts.snap
Comment (rozele) [inline packages/e2e-test-app/test/__snapshots__/PressableComponentTest.test.ts.snap:18]: Width shouldn't change...

**Comment by rozele** [comment]
Also noticed that this resolves #8420, which is pretty neat :)

**Comment by chiaramooney** [comment]
ping @chrisglein

**Comment by chiaramooney** [comment]
@rozele We should be able to merge this once the merge conflicts are resolved

**Comment by rozele** [comment]
> @rozele We should be able to merge this once the merge conflicts are resolved

@chiaramooney Done.

**Comment by chiaramooney** [comment]
@rozele Looks like one more merge conflict emerged end the E2E tests are failing.

**Comment by rozele** [comment]
@chiaramooney - The only thing I did was rebase and update the e2e test snapshots. I think this can now be merged.

**Review by acoates-ms — APPROVED** [review]

**Pull request merged**
